python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -771,19 +715,16 @@ def __init__(cls, name, bases, this_dict):
         # create reference in schema to the pluginclass
         this_schema._pluginclass = cls
 
-        # store generated proxy class
-        setattr(cls, '_proxyClass', proxyClass)
-
-        # register plugin class
-        if hasattr(cls, '_declared_property'):
-            # if we've not even declared this we don't want to use it!
-            if not cls._declared_property('hidden') or cls._declared_property('enable_plugin'):
-                allPlugins.add(cls, cls._category, getName(cls))
+        # if we've not even declared this we don't want to use it!
+        if not cls._declared_property('hidden') or cls._declared_property('enable_plugin'):
+            allPlugins.add(cls, cls._category, getName(cls))
 
-            # create a configuration unit for default values of object properties
-            if not cls._declared_property('hidden') or cls._declared_property('enable_config'):
-                this_schema.createDefaultConfig()
+        # create a configuration unit for default values of object properties
+        if not cls._declared_property('hidden') or cls._declared_property('enable_config'):
+            this_schema.createDefaultConfig()
 
+        # store generated proxy class
+        setattr(cls, '_proxyClass', GPIProxyClassFactory(name, cls))
```

Comment:
My plan was to have a class decorator called `add_proxy_and_export_to_gpi` (except obviously a better name) which would be used like:
``` python
@add_proxy_and_export_to_gpi
class MyGangaObject(GangaObject):
    _schema = Schema(Version(1, 0))
    _category = 'MyGangaObjects'
    _name = 'MyGangaObject'
```
so we would add that decorator to any class which we want to be proxyable. It would also do the `exportToGPI()` bit as well.

This way, with just that we could do
```python
MyGangaObject()
```
in the GPI and it would work.

All of the proxy stuff would then be removed from Objects.py as it would be imposed on it externally instead.